### PR TITLE
Use Tailwind CSS with Flask (example)

### PR DESCRIPTION
**Pull request by manzt**
I realized what I described in the long comment in #1 might be confusing. This PR is mostly a sketch of what I described there. It adds [Tailwind CSS](https://tailwindcss.com/) in `css`, which you can extend if desired with `css/styles.css` + `css/tailwind.config.js`. 

```bash
$ make # builds js and css static assets 
```

```bash
$ NODE_ENV=production make # builds js with minification and removes unused styles from final css
```

### CODE REVIEW

File: templates/viewer.html [hunk 1]
```diff
@@ -1,18 +1,10 @@
...
-	<script src="/static/bundle.js"></script>
-</body>
-
-</html>
+{% extends "base.html" %}
+{% block content %}
+<div id="WebGLContainer"></div>
+
+<script type="module">
+	import { startViewer, connectSocketInput } from '/static/viewer.js';
```
Comment (manzt) [inline templates/viewer.html:6]: With in a script module tag, you can import any module exports from your bundle(s).

File: js/build.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require('esbuild').build({
+    entryPoints: [
+        'src/viewer.js',
+        // other files you want to end up in /static
+    ],
+    format: 'esm',
+    outdir: '../static/',
+    bundle: true,
+    splitting: true,
+    minify: process.env.NODE_ENV === 'production', // only minify if production build
```
Comment (manzt) [inline js/build.js:10]: easier for debugging

File: css/tailwind.config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+	purge: [
+		'../templates/**/*.html'
```
Comment (manzt) [inline css/tailwind.config.js:3]: when `NODE_ENV=production`, `tailwind-cli` will crawl the `html` files in `templates` and "purge" any unused styles.

**Comment by pwinston** [comment]
Cool, thanks. Can't wait to try this. Seems like exactly the right idea. That would have taken me a while.

Unfortunately "main" is broken for me right now. I swear I tested it with the new bundling scheme, but maybe not? In latest you can run "python webmon.py" not inside napari as a test. And you can modify old version to do that as well.

Basically the web server starts, the viewer can pull down the html/javascript, it seems to make the socket connection. But no actual socketcio handlers gets call, on the JS side or the Python.

If you can try it let me know. I've banged on it for quite a while.

If I go back before the bundle that works. And the old FlaskTest works.

Easily could be something I broke though.

**Comment by pwinston** [comment]
@manzt awesome this is just what we needed.

Now I can setup a 2nd page and that really shows the WebUI can easily grow.

Thanks!

For the record it looks like this:

Screen Shot 2020-11-29 at 7 12 58 PM

File: js/build.js (hunk 2 above)
Comment (pwinston) [reply js/build.js]: I removed the `--minify` somewhere at one point when debugging. But this makes sense that you can have both modes.

File: templates/viewer.html (hunk 1 above)
Comment (pwinston) [reply templates/viewer.html:6]: Cool.

**Pull request merged**